astropy/coordinates/sky_coordinate.py

```diff
@@ -362,7 +363,41 @@ def _parse_inputs(self, args, kwargs):
 
         return valid_kwargs
 
-    def transform_to(self, frame):
+    def transforms(self, other, accepting_defaults=False):
```

Comment:
None of my tests cover mixing SkyCoords with data-bearing BaseCoordinateFrames.  Do we need some way of having a SkyCoord(frame=A) transform both SkyCoords and BaseCoordinateFrames into SkyCoord(frame=A) ?   (My `transforms()` function does not do this.)